gcsfs/retry.py

```diff
@@ -121,8 +111,15 @@ async def retry_request(func, retries=6, *args, **kwargs):
                 and e.code == 400
                 and "requester pays" in e.message
             ):
-                msg = "Bucket is requester pays. Set `requester_pays=True` when creating the GCSFileSystem."
+                msg = (
+                    "Bucket is requester pays. "
+                    "Set `requester_pays=True` when creating the GCSFileSystem."
+                )
                 raise ValueError(msg) from e
+            # Special test for 404 to avoid retrying the request
+            if isinstance(e, aiohttp.client_exceptions.ClientError) and e.code == 404:
+                logger.debug("Request returned 404, no retries.")
+                raise e
```

Comment:
Just `raise` alone

Reply:
Just to be consistent with the rest of the file, see this [line](https://github.com/dask/gcsfs/pull/406/files#diff-d89213d3b1e29d457dc7aed940ccb379f48e8ef45f34495e2ab3934e39124534L130) and this [line](https://github.com/dask/gcsfs/pull/406/files#diff-d89213d3b1e29d457dc7aed940ccb379f48e8ef45f34495e2ab3934e39124534L135)